packages/wdio-config/src/dataproviders.js

```diff
@@ -0,0 +1,42 @@
+import { safeRequire } from '@wdio/utils'
+import ConfigParser from './lib/ConfigParser'
+
+/**
+ * Loads the data providers from the files to memory
+ * @param  {Object} dataProviderFilesPaths  List of files with data provider injections
+ */
+export function initializeDataProviders (dataProviderFilesPaths) {
+    const dataProviderFiles =  ConfigParser.getFilePaths(dataProviderFilesPaths)
+    const dataProvidersMap = {}
+    let currentDataProviderFile = ''
+
+    global.dataProvider = function (specFilePath, dataSet) {
+        if (!Array.isArray(dataSet)) {
+            throw new Error(`The data set passed from the file "${currentDataProviderFile}" for the spec file path "${specFilePath}" is not an array.`)
+        }
+
+        let specFiles =  ConfigParser.getFilePaths(specFilePath)
+        specFiles.forEach((specFile) => {
+            const specFileDataProvider = dataProvidersMap[specFile]
+            if(typeof specFileDataProvider === 'undefined') {
```

Comment:
```suggestion
            if (typeof specFileDataProvider === 'undefined') {
```

there should be an eslint rule for that